### PR TITLE
[#1356] feat(server): improve expired buffers metric and log

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?

improve expired buffers metric and log

Once the expired buffer happened, it means the server or client may have problems, 
we should find out the related apps by the log shown in the server

### Why are the changes needed?

For #1356 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests

### CODE REVIEW

File: server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java [hunk 1]
```diff
@@ -478,22 +478,27 @@ public long requireBuffer(
...
+          new PreAllocatedBufferInfo(appId, requireId, System.currentTimeMillis(), requireSize));
       return requireId;
     } else {
       LOG.error("Failed to require buffer, require size: {}", requireSize);
       throw new NoBufferException("No Buffer For Regular Partition, requireSize: " + requireSize);
     }
   }
 
+  public long requireBuffer(int requireSize) {
+    // appId of EMPTY means the client uses the old version that should be upgraded.
```
Comment (jerqi) [inline server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java:498]: Could you elaborate more?

Comment (zuston) [reply server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java:498]: Miss something?

Comment (zuston) [reply server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java:498]: The appId of requireBuffer proto is introduce in #471 . So if using the older version client, the appId may be empty.

Comment (jerqi) [reply server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java:498]: OK. Maybe we should add a `todo` for this.

Comment (zuston) [reply server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java:498]: But I found the current server has been not compatible with the version of #471 merged. That means we could remove this currently, let's do this in other PR.

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zuston

**Pull request merged**